### DISCUSSION TITLE
AI DECK programming without the rest of the drone. Is it possible?

### DISCUSSION THREAD

**Discussion by Alejandroperezm** [bitcraze/discussions#1296] [Q&A]
Hello,

 I have bought 2 AI DECK 1.1, I want to try the camera and the AI capabilities of the GAP8 but I don´t have the rest of the dron, as i only care about the AI DECK extension. Is there anyway to use the AI deck without the rest of the Crazyflie dron? I want to power the AI DECK, connect it to PC and program the GAP8 microcontroller. 

Is there any guide or clue you can give me? Any help is very much appreciated. I have read all the datasheets and files in the web site but im unable to find any substancial information.

Thank you.

**Comment by evoggy** [comment]
Hi,

This is possible but a bit tricky and requires extra hardware. You will probably need an ESP PROG (for programming the ESP32) and a JTAG (we use an Olimex one) for programming the GAP8. You can connect the ESP PROG to the expansion connector getting both UART and the EN/BOOT signals as well as power and then connect the JTAG directly to the AI deck.

You can then use UART for communicating with the ESP32, maybe using the same protocol (CPX) as we use on the Crazyflie. Have a look at the [SerialDriver](https://github.com/bitcraze/crazyflie-lib-python/blob/master/cflib/crtp/serialdriver.py#L56) in the library, this might be able to be used directly with the AI deck.

For reference:
* [CPX documentation](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/cpx/)
* [AI deck schematics](https://www.bitcraze.io/documentation/hardware/ai_deck_1_1/ai-deck-revc.pdf)
* [Expansion port pintout](https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#expansion-port-pinout)

**Reply by Alejandroperezm** [reply]
Ok thank you very much for the information, I was also thinking about the ESP PROG for JTAG connection. But for  the UART connection i still don´t understand fully.

From the schematics of the AI deck: [https://www.bitcraze.io/documentation/hardware/ai_deck_1_1/ai-deck-revc.pdf](url)
![nina module in AI DECK](https://github.com/bitcraze/discussions/assets/99073479/4276dc16-f110-45ea-8f8b-48166f15006d)
We have to connect both NINA_UART_RX and TX, as well as the BOOT pins to the program interface of the ESP-PROG board, this will allow us to communicate with the ESP32 on the AI DECK, and let us monitor the program through PC viewer? Is that correct? Do we actually need the ESP32 module if we already have the JTAG interface connected to PC through the ESP-PROG? (This is what i don´t understand)

Another question that comes to my mind, you haven´t talked about powering the AI Deck. Does the JTAG connection already provide the power to the GAP8 without no need for connecting VCOM and GND form pin jumpers?

**Reply by evoggy** [reply]
You need to use the ESP32 to get the data from the GAP8 (or to use the GAP8 UART). The JTAG power is only used for sensing the voltage, you need to power the deck on the expansion connector.
